File: frontend/src/lib/types/DTO/helpers/SprintMemberTasksDataDTO.ts
import { z } from "zod";
import { MemberTasksItemSchema } from "./MemberTasksItemDTO";

export const SprintMemberTasksDataItemSchema = z.object({
    sprintName:    z.string(),
    tasksByMember: z.array(MemberTasksItemSchema),
});

export const SprintMemberTasksDataSchema = z.array(SprintMemberTasksDataItemSchema);
export type SprintMemberTasksDataDTO = z.infer<typeof SprintMemberTasksDataSchema>;

File: lib/src/main/java/DTO/helpers/SprintMemberTasksDataDTO.java
package DTO.helpers;

import java.util.List;

/**
 * DTO for the "Completed Tasks by Team Member" chart.
 * Contains data for one sprint, including a list of completed tasks by each relevant member.
 */

public record SprintMemberTasksDataDTO(
        String sprintName,
        List<MemberTasksItemDTO> tasksByMember
) {
}
